JavaScript development/Frameworks/React.md

```diff
@@ -0,0 +1,389 @@
+
+### File organization and naming
+- as it is a good practice to divide React components as controllers and presentational components, you should place them in separate folders (example: **containers** and **components**)
+
+```
+...
+├── components
+├── containers 
+...
+```
+
+- each component should have it's own folder containing it's logic in a JSX file and it's stylesheet file.
+ (If the app is (or becomes) more complex the components can be placed in additional logically separated sub-folders for better readability and organization).
+- use **camelCase** for general folder naming and **PacalCase** for React components, their stylesheet files and their containing folders 
+- each component type's root folder should have an **index.jsx** file which exports all components, so you can later import them from a single place
+
+```
+...
+├── components
+    ├── LoginContent
+        └── LoginContent.jsx
+        └── LoginContent.scss
+    ├── LoginHeader
+        └── LoginHeader.jsx
+        └── LoginHeader.scss
+    └── index.jsx
+├── containers 
+    ├── Login
+        └── Login.jsx
+        └── Login.scss 
+    └── index.jsx
+...
+```
+
+### Component Declaration
+- don't use React.createClass
+- use ES2015 **class syntax** for component declaration
+
+``` javascript
+// bad
+const ComponentName = React.createClass({
+  // ...
+  render () {
+    return (
+      <div>
+        {this.componentState.message}
+      </div>
+    );
+  }
+});
+    
+// good
+import React, {Component} from 'react';
+
+class ComponentName extends Component {
+  // ...
+  render () {
+    return (
+      <div>
+        {this.componentState.message}
+      </div>
+    );
+  }
+}
+    
+export default ComponentName;
+```
+
+- if components don't have state, use **pure functions**:
+
+``` javascript
+import React from 'react';
+
+const ComponentName = ({message}) => {
+  return (
+    <div>
+      {message}
+    </div>
+  );
+};
+```
+
+### Imports
+- import specific React modules, such as **Component**, to avoid accessing them with the React namespace every time
+
+``` javascript
+// not that good
+import React from 'react';
+
+class ComponentName extends React.Component {
+  // ...
+}
+
+// good
+import React, {Component} from 'react';
+
+class ComponentName extends Component {
+  // ...
+}
+    
+export default ComponentName;
+```
+
+### Exports
+- each JSX file should have a single exported component
+ 
+``` javascript
+export default class ComponentName extends Component {
+  // ...
+}
+    
+// or
+
+class ComponentName extends Component {
+  // ...
+}
+    
+export default ComponentName;
+```
+
+### Props Passing
+
+- use camelCase for props
+- use double quotes for JSX attribute values if you are passing them a *string*
+- use self-closing tags if the element doesn't have children
+
+``` javascript
+class ComponentName extends Component {
+    // ...
+    render () {
+        return (
+          <ChildComponent
```

Comment:
You should add a separate part where you describe how to format components - if it doesn't fit into one row, put every property in a separate row and the closing `>` or `/>` in its own row.